src/main/java/util/TickingTimer.java

```diff
@@ -158,4 +158,13 @@ public boolean isStarted() {
         return started;
     }
 
+    /**
+     * Changes the timer's period and restart the timer based on the new period.
+     * @param periodInSec : the amount of time in second before trigger of the timer.
+     */
+    public synchronized void restartTimerBasedOnNewPeriod(int periodInSec){
```

Comment:
The `BasedOnNewPeriod` can be omitted from the name because the presence of the parameter implies a new period will be used. It's good to write explanatory names, but we should try to keep them no longer than necessary.